File: app/src/main/java/ru/leushinilya/loftmoney/cells/Item.java
package ru.leushinilya.loftmoney.cells;

import ru.leushinilya.loftmoney.remote.RemoteItem;

public class Item {

    private final String name, price;
//    0 - expense, 1 -income
    int type;

    public Item(String name, String price, int type) {
        this.name = name;
        this.price = price;
        this.type = type;

    }

    public String getName() {
        return name;
    }

    public String getPrice() {
        return price;
    }

    public int getType() {
        return type;
    }

    public static Item getInstance(RemoteItem remoteItem) {
        String remoteItemName = remoteItem.getName();
        String remoteItemPrice = "" + remoteItem.getPrice();
        int remoteItemType;
        if(remoteItem.getType().equals("expense")) remoteItemType = 0; else remoteItemType = 1;
        return new Item(remoteItemName, remoteItemPrice, remoteItemType);
    }
}

File: app/src/main/java/ru/leushinilya/loftmoney/cells/Item.kt
package ru.leushinilya.loftmoney.cells

import ru.leushinilya.loftmoney.data.remote.RemoteItem

data class Item(
    val id: String,
    val name: String,
    val price: String, //    0 - expense, 1 -income
    var type: Int
) {
    companion object {
        fun getInstance(remoteItem: RemoteItem): Item {
            val remoteItemId = remoteItem.id
            val remoteItemName = remoteItem.name
            val remoteItemPrice = "" + remoteItem.price
            val remoteItemType: Int
            remoteItemType = if (remoteItem.type == "expense") 0 else 1
            return Item(remoteItemId, remoteItemName, remoteItemPrice, remoteItemType)
        }
    }
}